pkg/operator/sync.go

```diff
@@ -687,6 +687,19 @@ func (optr *Operator) syncMachineConfigPools(config *renderConfig) error {
 			return err
 		}
 		p := resourceread.ReadSecretV1OrDie(userdataBytes)
+
+		// Work around https://github.com/kubernetes/kubernetes/issues/3030 and https://github.com/kubernetes/kubernetes/issues/80609
+		pool.APIVersion = mcfgv1.GroupVersion.String()
+		pool.Kind = "MachineConfigPool"
```

Comment:
I'm :+1: on this, just curious on the choice of MCP as the "owner" here. Technically multiple MCPs can be using the same secret (although we don't support custom pool booting generally so that's mostly a moot point)

I think master and worker MCPs can be deleted but will be immediately recreated assuming the MCO is operating, so I don't foresee that as an issue

I don't suppose the owner can be set to the MCO operator itself?

Reply:
> Technically multiple MCPs can be using the same secret...

I thought this code was creating `${POOL_NAME}-user-data-managed`, and I'm not seeing how there could be multiple pools using _that_ same secret.  I could see multiple pools using the same... whatever the MCO is pulling from to create these managed Secrets.  But I'm not setting `ownerReferences` on that source.

> I don't suppose the owner can be set to the MCO operator itself?

No, because `ownerReferences` docs say [cross-namespace owner references are disallowed by design][1], and the user-data secrets are in `openshift-machine-api`, while the MCO Pod and Deployment are in `openshift-machine-config-operator`.  The pools are cluster-scoped, and from those same upstream docs:

> Namespaced dependents can specify cluster-scoped or namespaced owners.

[1]: https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/#owner-references-in-object-specifications

Reply:
Hmm, weird, I just noticed that the selector is selecting for all pools. Custom pools (and nodes) should be using worker secrets, and we don't explicitly support booting into custom pools, so they shouldn't ever get created by the MCO, but I guess since nobody is using them there isn't really an issue.

I'm :+1: on having his as is but I'd have to re-check on how that works and if it's actually generating any use-able secrets for other pools

Reply:
If nobody consumes `*-user-data-managed` Secrets, I'm +1 on dropping them entirely, instead of updating them to set `ownerReferences`  ;)

Reply:
Sorry, to clarify, there is a singular use case where user-data-managed is used: bare metal IPI

I'm not sure why they didn't opt to use the worker-user-data secret. I didn't know about this until we did the FIPS cert rotations